lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/resourcemanager/ResourceManagerStatus.java

```diff
@@ -98,23 +109,30 @@ public synchronized void setRunning() {
   }
 
   /**
-   * @return idle, if there are no outstanding requests or allocations. Not idle else.
+   * Driver is idle if, regardless of status, it has no evaluators allocated and no pending container requests.
+   * @return true if the driver can be considered idle, false otherwise.
```

Comment:
We should probably add a short example of when this happens and how REEF uses this.